examples/react-router/src/App.js

```diff
@@ -23,29 +23,26 @@ const searchStateToUrl = (props, searchState) =>
 const urlToSearchState = location => qs.parse(location.search.slice(1));
 
 class App extends Component {
-  constructor(props) {
-    super(props);
-
-    this.state = {
-      searchState: urlToSearchState(props.location),
-    };
-  }
+  state = {
+    searchState: urlToSearchState(this.props.location),
+  };
 
-  componentWillReceiveProps(props) {
-    // @TODO: derived state
-    if (props.location !== this.props.location) {
-      this.setState({ searchState: urlToSearchState(props.location) });
+  componentDidUpdate(prevProps) {
```

Comment:
in the other example you used derived state

Reply:
React Router v3 behaves differently from React Router 5. If you try the example they should react the same way.